projects/nx-verdaccio/src/plugin/targets/environment.targets.ts

```diff
@@ -114,13 +126,31 @@ export function getEnvTargets(
     },
     // runs env-bootstrap-env, install-env and stop-verdaccio
     [TARGET_ENVIRONMENT_SETUP]: {
+      // list of inputs that all subsequent tasks depend on
+      inputs: [
+        '{projectRoot}/project.json',
+        {
+          runtime: 'node --version',
+        },
+        {
+          runtime: 'npm --version',
+        },
+        {
+          externalDependencies: ['verdaccio'],
+        },
+        // depends on underlying project being e2e tested and its own dependencies
+        // ! it's important that implicitDependencies are correctly configured between this project and the project being tested
+        // '^build-artifacts',
+        '^production',
+      ],
       outputs: [
-        '{options.environmentRoot}/node_modules',
-        '{options.environmentRoot}/package.json',
+        `{options.environmentRoot}/${VERDACCIO_STORAGE_DIR}`,
```

Comment:
As we don't need the server running this folder should not be cached. 
When the test executes only the node_modules folder is needed.

Reply:
I thought the same, but we still need to find out if caching the node_modules is a good choice. 
Chances are high it will backfire with time and larger projects.
We should try using Nx replay and see how it goes before merging.

Reply:
> with time and larger projects
I thought about it. The way how I think about it ATM is, downloading the packages from NPM every time the same e2e runs vs 1 time and later from cache... 

I just had no time to do proper benchmarks for the cache download time etc.

My suggestion would be to have a new PR where we also have some benchmarks in the code-pushup plugin.
WDYT? Would open a new issue for this.

Reply:
we can keep it simple for now, and a workspace with more projects and more dependencies would be helpful to mock more realistic conditions.

Reply:
Good when you opened the new issue please close this conversation

Reply:
I just added a task in #33